packages/framework/presence/src/test/notificationsManager.vitest.ts

```diff
@@ -0,0 +1,114 @@
+			let logger: EventAndErrorTrackingLogger;
+			const initialTime = 1000;
+			let clock: SinonFakeTimers;
+			let presence: ReturnType<typeof createPresenceManager>;
+
+			beforeAll(async () => {
+				clock = useFakeTimers();
+			});
+
+			beforeEach(() => {
+				logger = new EventAndErrorTrackingLogger();
+				runtime = new MockRuntimeSignalSnapshotter(logger);
+
+				// We are configuring the runtime to be in a connected state, so ensure it looks connected
+				runtime.connected = true;
+
+				// Note that while the initialTime is set to 1000, the prepareConnectedPresence call advances
+				// it to 1010 so all tests start at that time.
+				clock.setSystemTime(initialTime);
+
+				// Disable submitting signals with a dummy function. This ensures we don't capture signals from
+				// test setup, like the prepareConnectedPresence call.
+				const submitSignalOriginal = runtime.submitSignal;
+				runtime.submitSignal = () => {};
+
+				// Set up the presence connection
+				presence = prepareConnectedPresence(runtime, "sessionId-2", "client2", clock, logger);
+
+				// Restore the submiSignal function
+				runtime.submitSignal = submitSignalOriginal;
+			});
+
+			afterEach(() => {
+				clock.reset();
+			});
+
+			afterAll(() => {
+				clock.restore();
+			});
+
+			it("notification signals are sent immediately", async () => {
+				// Configure a notifications workspaces
+				// eslint-disable-next-line @typescript-eslint/ban-types
+				const notificationsWorkspace: PresenceNotifications<{}> = presence.getNotifications(
+					"name:testNotificationWorkspace",
+					{},
+				);
+
+				notificationsWorkspace.add(
+					"testEvents",
+					Notifications<
+						// Below explicit generic specification should not be required.
+						{
+							newId: (id: number) => void;
+						},
+						"testEvents"
+					>(
+						// A default handler is not required
+						{},
+					),
+				);
+
+				const { testEvents } = notificationsWorkspace.props;
+
+				clock.tick(40); // Time is now 1050
+				// SIGNAL #1
+				testEvents.emit.broadcast("newId", 77);
+
+				clock.tick(10); // Time is now 1060
+				// SIGNAL #2
+				testEvents.emit.broadcast("newId", 88);
+
+				expect(runtime.submittedSignals).toHaveLength(2);
+
+				// Verify first signal set the newId to 77
+				let signal = runtime.submittedSignals[0];
+				expect(
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+					(signal?.[1] as any).data["n:name:testNotificationWorkspace"].testEvents[
+						"sessionId-2"
+					].value.args,
+				).toEqual([77]);
+
+				// Verify first signal set the newId to 88
+				signal = runtime.submittedSignals[1];
+				expect(
+					// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+					(signal?.[1] as any).data["n:name:testNotificationWorkspace"].testEvents[
+						"sessionId-2"
+					].value.args,
+				).toEqual([88]);
```

Comment:
Philosophically:
Should we have / bother with these checks of signal content when we have snapshots? (I guess probably yes and here we are basically saying these parts of the snapshot should be harder to change accidentally.)
But if we have these, how much do we care about the other nitty gritty? Are these (and maybe few others) the only value checks that should be in place? (Big snapshots are a concern for me in that when you check everything you are over-validating and make it hard to change tangential aspects. There will be a lot of noise from such tangential changes, and it can be easy to miss that something important was changed that shouldn't have been.)

NotificationsManager or any Value Manager isn't responsible for the signal really. What matters is that others receive update in the right form with the right data. The signal checks done elsewhere were because that piece (presenceDataStoreManager.ts) is the component that does the signaling. I do think that we should have some higher-order unit integration tests that check full signals, but I am not thinking that should be the most common. Perhaps we can build support for:
1. checking that a signal has certain parts (signalsExpected array could be an object or a function array - when function we call it with the current signal content to be verified (pieces are verified)) and/or
2. signal generated by A and processed by B yields the expected update/notification event to caller.